bundles/org.openhab.binding.buienradar/src/main/java/org/openhab/binding/buienradar/internal/buienradarapi/BuienradarPredictionAPI.java

```diff
@@ -0,0 +1,152 @@
+/**
+ * Copyright (c) 2010-2019 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.buienradar.internal.buienradarapi;
+
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.smarthome.io.net.http.HttpUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link BuienradarPredictionAPI} class implements the methods for retrieving results from the buienradar.nl
+ * service.
+ *
+ * @author Edwin de Jong - Initial contribution
+ */
+@NonNullByDefault
+public class BuienradarPredictionAPI implements PredictionAPI {
+    private static final ZoneId ZONE_AMSTERDAM = ZoneId.of("Europe/Amsterdam");
+
+    private static final String BASE_ADDRESS = "https://gpsgadget.buienradar.nl/data/raintext";
+
+    private static final int TIMEOUT = 15000;
```

Comment:
15 seconds is a very long timeout. Especially considering this runs on a limited scheduled pool. So if there is something wrong with the service and it times. So maybe 1, 2 or 3 seconds is a better option. Also prefer to use the unit in the constant name: `TIMEOUT_MILLISECONDS`

Reply:
The buienradar service is sometimes very slow, that's why I increased it to 15000. I'll place it back to 3 seconds for now and see how it runs.

(btw, seems like a good incentive to move these kind of HTTP calls to their own HTTP client pool and return a Future instead. Haven't done professional programming in Java for a while, so I don't know how it compares with Scala though.)

Reply:
The httpclient that we use is capable of async requests with timeouts (internally hopefully with java.nio implemented if not at least it is using its own thread pool). You might indeed want to use that instead

Reply:
The common usage of the httpclient is wrapped many times in `HttpUtil` without creating an orthogonal abstraction (see `org.eclipse.smarthome.io.net.http.HttpUtil.executeUrlAndGetReponse(String, String, Properties, InputStream, String, int, String, Integer, String, String, String)`). So I'd have to implement all this logic again, but then in an asynchronous manner. 

So, in a certain way I cannot use the underlying HTTP client without reimplementing that logic, since I'd break proxy support, expected debug logging, auth headers and shared HTTP client pools (although I think that's not applicable here, since we're sharing via Jetty).

Also, certain logic wrt proxying is hidden (`org.eclipse.smarthome.io.net.http.HttpUtil.prepareProxyParams()`), so even if I would copy it,  I might break forward compatibility, since this method could be changed in the future.

Reply:
Oh, I don't mean to use `org.eclipse.smarthome.io.net.http.HttpUtil`. I have never done that and don't understand its use case to be honest. There is a `HttpClientFactory`. Get that as a service in your thing factory (or discovery services) and pass it to where you need it. Let the factory create a httpclient for you and voila

Reply:
Aha, well, I'll migrate to that in the next PR with auto discovery. Out of time for this week... I picked up the `HttpUtil` from elsewhere (forgot which binding), thinking that this is 'normal' path. Seeing the proxy code using properties, I assumed other avenues are not wanted. There are *so* *many* *HTTP* client libraries out there... Simply don't know which one to use or how to even find information on which one is preferable.